Compile warnings in VS 2022 17.8.0 - stdext::make_checked_array_iterator std::make_unchecked_array_iteratorbeeing deprecated

I`m seeing compiler warnings since the latest VS 2022 update (17.8.0) on Windows 11 with the latest cryptopp 8.9

`17:30:31:685	1>X:\cryptopp\integer.cpp(3061,12): warning C4996: 'stdext::make_checked_array_iterator': warning STL4043: stdext::checked_array_iterator, stdext::unchecked_array_iterator, and related factory functions are non-Standard extensions and will be removed in the future. std::span (since C++20) and gsl::span can be used instead. You can define _SILENCE_STDEXT_ARR_ITERS_DEPRECATION_WARNING or _SILENCE_ALL_MS_EXT_DEPRECATION_WARNINGS to suppress this warning.
17:30:31:685	1>(compiling source file '/integer.cpp')

17:30:32:401	1>X:\cryptopp\zdeflate.cpp(422,155): warning C4996: 'stdext::unchecked_array_iterator<const CryptoPP::byte *>::operator -': warning STL4043: stdext::checked_array_iterator, stdext::unchecked_array_iterator, and related factory functions are non-Standard extensions and will be removed in the future. std::span (since C++20) and gsl::span can be used instead. You can define _SILENCE_STDEXT_ARR_ITERS_DEPRECATION_WARNING or _SILENCE_ALL_MS_EXT_DEPRECATION_WARNINGS to suppress this warning.
`

It appears loosen from the platformSDKiand loosen from the C++ standard

Tried PlatformSDK:
- 10.0.19041.0
- 10.0.22000.0

C++ Language standard:
- ISO C++17 Standard (/std:c++17)
- ISO C++20 Standard (/std:c++20)

--- a/integer.cpp
+++ b/integer.cpp
@@ -3056,12 +3056,7 @@ Integer::Integer(const byte *encodedInteger, size_t byteCount, Signedness s, Byt
 	else
 	{
 		SecByteBlock block(byteCount);
-#if (CRYPTOPP_MSC_VERSION >= 1500)
-		std::reverse_copy(encodedInteger, encodedInteger+byteCount,
-			stdext::make_checked_array_iterator(block.begin(), block.size()));
-#else
 		std::reverse_copy(encodedInteger, encodedInteger+byteCount, block.begin());
-#endif
 		Decode(block.begin(), block.size(), s);
 		return;
 	}

--- a/integer.cpp
+++ b/integer.cpp
@@ -3056,7 +3056,12 @@ Integer::Integer(const byte *encodedInteger, size_t byteCount, Signedness s, Byt
 	else
 	{
 		SecByteBlock block(byteCount);
+#if (CRYPTOPP_MSC_VERSION >= 1500) && (CRYPTOPP_MSC_VERSION < 1938)
+		std::reverse_copy(encodedInteger, encodedInteger+byteCount,
+			stdext::make_checked_array_iterator(block.begin(), block.size()));
+#else
 		std::reverse_copy(encodedInteger, encodedInteger+byteCount, block.begin());
+#endif
 		Decode(block.begin(), block.size(), s);
 		return;
 	}

--- a/zdeflate.cpp
+++ b/zdeflate.cpp
@@ -412,17 +412,7 @@ unsigned int Deflator::LongestMatch(unsigned int &bestMatch) const
 		if (scan[bestLength-1] == match[bestLength-1] && scan[bestLength] == match[bestLength] && scan[0] == match[0] && scan[1] == match[1])
 		{
 			CRYPTOPP_ASSERT(scan[2] == match[2]);
-			unsigned int len = (unsigned int)(
-#if defined(_STDEXT_BEGIN) && !(defined(CRYPTOPP_MSC_VERSION) && (CRYPTOPP_MSC_VERSION < 1400 || CRYPTOPP_MSC_VERSION >= 1600)) && !defined(_STLPORT_VERSION)
-				stdext::unchecked_mismatch
-#else
-				std::mismatch
-#endif
-#if CRYPTOPP_MSC_VERSION >= 1600
-				(stdext::make_unchecked_array_iterator(scan)+3, stdext::make_unchecked_array_iterator(scanEnd), stdext::make_unchecked_array_iterator(match)+3).first - stdext::make_unchecked_array_iterator(scan));
-#else
-				(scan+3, scanEnd, match+3).first - scan);
-#endif
+			unsigned int len = (unsigned int)(std::mismatch(scan+3, scanEnd, match+3).first - scan);
 			CRYPTOPP_ASSERT(len != bestLength);
 			if (len > bestLength)
 			{

--- a/zdeflate.cpp
+++ b/zdeflate.cpp
@@ -412,7 +412,12 @@ unsigned int Deflator::LongestMatch(unsigned int &bestMatch) const
 		if (scan[bestLength-1] == match[bestLength-1] && scan[bestLength] == match[bestLength] && scan[0] == match[0] && scan[1] == match[1])
 		{
 			CRYPTOPP_ASSERT(scan[2] == match[2]);
-			unsigned int len = (unsigned int)(std::mismatch(scan+3, scanEnd, match+3).first - scan);
+			unsigned int len = (unsigned int)(std::mismatch
+#if (CRYPTOPP_MSC_VERSION >= 1600) && (CRYPTOPP_MSC_VERSION < 1938)
+				(stdext::make_unchecked_array_iterator(scan)+3, stdext::make_unchecked_array_iterator(scanEnd), stdext::make_unchecked_array_iterator(match)+3).first - stdext::make_unchecked_array_iterator(scan));
+#else
+				(scan+3, scanEnd, match+3).first - scan);
+#endif
 			CRYPTOPP_ASSERT(len != bestLength);
 			if (len > bestLength)
 			{

